ethcore/src/header.rs

```diff
@@ -186,11 +186,12 @@ impl Header {
 	/// Get the hash of the header excluding the seal
 	pub fn bare_hash(&self) -> H256 {
 		let mut hash = self.bare_hash.borrow_mut();
-		match &mut *hash {
+		match hash.deref_mut() {
```

Comment:
`Deref`/`DerefMut` are compiler magic, like most of the traits in `std::ops`. You can also explicitly call `Add` implementations using `a.add(b)`, but it is preferred to use the operator itself.

This isn't as useful for match, but for function arguments, deref coercions are applied repeatedly until a type which fulfills pending obligations is found -- so it is strictly more flexible than calling a single `deref[_mut]()`. This could allow the same code to continue working if the type were changed from `RefCell<Option<H256>>` to, say, a `RefCell<Box<Option<H256>>>`. For the sake of consistency people continue to use the `&mut *x` pattern in matches.